sdcm/cluster.py

```diff
@@ -3461,6 +3465,40 @@ def restore_keyspace(self, backup_data):
         for node in self.nodes:
             node.run_nodetool('repair')
 
+    def decommission(self, node):
+        def get_node_ip_list(verification_node):
+            try:
+                ip_node_list = []
+                status = self.get_nodetool_status(verification_node)
+                for nodes_ips in status.values():
+                    ip_node_list.extend(nodes_ips.keys())
+                return ip_node_list
+            except Exception as details:  # pylint: disable=broad-except
+                LOGGER.error(str(details))
+                return None
+
+        target_node_ip = node.ip_address
+        node.run_nodetool("decommission")
+        verification_node = random.choice(self.nodes)
+        node_ip_list = get_node_ip_list(verification_node)
+        while verification_node == node or node_ip_list is None:
+            verification_node = random.choice(self.nodes)
+            node_ip_list = get_node_ip_list(verification_node)
+
+        if target_node_ip in node_ip_list:
+            cluster_status = self.get_nodetool_status(verification_node)
+            error_msg = ('Node that was decommissioned %s still in the cluster. '
+                         'Cluster status info: %s' % (node,
+                                                      cluster_status))
+
+            LOGGER.error('Decommission %s FAIL', node)
```

Comment:
You can combine the two log messages into one message
```
LOGGER.error(f'Decommission {node} FAIL\n{error_msg}',)

Reply:
maybe, did looked into this much, since I'm just moving this whole section from side to side.
and it's not exactly equivalent in all cases, depend on the logger formatting.

also using f-string in loggers, make them slower, in case this is a level that if going to be filtered, f-strings are not lazy.